app/controllers/comments_controller.rb

```diff
@@ -3,13 +3,19 @@ def create
      @post = Post.find(params[:post_id])
      @comment = current_user.comments.build(comment_params)
      @comment.post_id = @post.id
-     @user = current_user
+     @topic = Topic.find(params[:topic_id])
+     authorize @comment
+     
      if @comment.save
       flash[:notice] = "Comment was saved."
      else
       flash[:error] = "There was an error saving the comment. Please try again."
      end
-     redirect_to [@post.topic, @post]
+     redirect_to [@topic, @post]
+  end
+
+  def new
+    authorize @comment
   end
```

Comment:
so u dont need a new action here. u have the create and thats all u need. u are already authorizing its creation in the create method.